extensions/github-repository-search/src/index.tsx

```diff
@@ -11,55 +11,113 @@ import {
   Toast,
 } from "@raycast/api";
 import { useState } from "react";
-import { Repository } from "./types";
+import { Repository, UserDataResponse } from "./types";
 import { useDebounce } from "use-debounce";
 import { useRepositories } from "./useRepositories";
 import { clearVisitedRepositories, useVisitedRepositories } from "./useVisitedRepositories";
 import { getAccessoryTitle, getIcon, getSubtitle } from "./utils";
 import { useRepositoryReleases } from "./useRepositoryReleases";
 import { OpenInWebIDEAction } from "./website";
+import { useUserData } from "./useGithubUser";
 
 export default function Command() {
   const [searchText, setSearchText] = useState<string>();
+  const [searchFilter, setSearchFilter] = useState<string>("");
   const [debouncedSearchText] = useDebounce(searchText, 200);
-  const { data, error, isLoading: isLoadingRepositories } = useRepositories(debouncedSearchText);
+  const {
+    data: repositories,
+    error: searchRepositoriesError,
+    isLoading: isLoadingRepositories,
+  } = useRepositories(debouncedSearchText, searchFilter);
+  const { data: userData, error: fetchUserDataError } = useUserData();
+
   const {
     repositories: visitedRepositories,
     visitRepository,
     isLoading: isLoadingVisitedRepositories,
   } = useVisitedRepositories();
 
-  if (error) {
+  if (searchRepositoriesError) {
     showToast({
       style: Toast.Style.Failure,
       title: "Failed searching repositories",
-      message: error instanceof Error ? error.message : String(error),
+      message:
+        searchRepositoriesError instanceof Error ? searchRepositoriesError.message : String(searchRepositoriesError),
+    });
+  }
+  if (fetchUserDataError) {
+    showToast({
+      style: Toast.Style.Failure,
+      title: "Failed fetching user and organizations",
+      message: fetchUserDataError instanceof Error ? fetchUserDataError.message : String(fetchUserDataError),
     });
   }
 
   const isLoading = searchText !== debouncedSearchText || isLoadingVisitedRepositories || isLoadingRepositories;
 
   return (
-    <List isLoading={isLoading} onSearchTextChange={setSearchText}>
+    <List
+      isLoading={isLoading}
+      onSearchTextChange={setSearchText}
+      searchBarAccessory={<FilterDropdown userData={userData} onFilterChange={setSearchFilter} />}
+    >
       <List.Section
         title="Visited Repositories"
         subtitle={visitedRepositories ? String(visitedRepositories.length) : undefined}
       >
         {visitedRepositories
-          ?.filter((r) => r.nameWithOwner.includes(searchText ?? ""))
+          ?.filter(
+            (r) =>
+              r.nameWithOwner.includes(searchText ?? "") &&
+              r.nameWithOwner.match(
+                // Converting query filter string to regexp:
+                `${searchFilter?.replaceAll(/org:|user:/g, "").replaceAll(" ", "|")}/.*`
+              )
```

Comment:
Shouldn't that be a disjunction?
```suggestion
              r.nameWithOwner.includes(searchText ?? "") ||
              r.nameWithOwner.match(
                // Converting query filter string to regexp:
                `${searchFilter?.replaceAll(/org:|user:/g, "").replaceAll(" ", "|")}/.*`
              )
```

Reply:
Why?
The second filter just targets the owner part of the repository id. Even if you get a hit with the first filter on something I think it should still filter the results by whatever you have selected in the dropdown